src/app/domain/backup/services/uploadMedias.ts

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable promise/always-return */
+import Minilog from '@cozy/minilog'
+import { CameraRoll } from '@react-native-camera-roll/camera-roll'
+import { Platform } from 'react-native'
+import RNBackgroundUpload, {
+  UploadOptions
+} from 'react-native-background-upload'
+import RNFileSystem from 'react-native-fs'
+
+import { setMediaAsBackuped } from '/app/domain/backup/services/manageLocalBackupConfig'
+import { Media } from '/app/domain/backup/models/Media'
+
+import type CozyClient from 'cozy-client'
+
+const log = Minilog('💿 Backup')
+
+interface UploadMediaResult {
+  success: boolean
+  data?: object
+}
+
+export const uploadMedias = async (
+  client: CozyClient,
+  backupFolderId: string,
+  mediasToUpload: Media[]
+): Promise<boolean> => {
+  for (const mediaToUpload of mediasToUpload) {
+    try {
+      const uploadUrl = getDriveUrl(client, backupFolderId, mediaToUpload)
+
+      const { success } = await uploadMedia(client, uploadUrl, mediaToUpload)
+
+      if (success) {
+        log.debug(`✅ ${mediaToUpload.name} uploaded`)
+        await setMediaAsBackuped(client, mediaToUpload)
+      } else {
+        log.debug(`❌ ${mediaToUpload.name} not uploaded`)
+      }
+    } catch (e) {
+      log.debug(`❌ ${mediaToUpload.name} not uploaded`)
+      log.debug(e)
+    }
+  }
+
+  return true
+}
+
+const uploadMedia = async (
+  client: CozyClient,
+  uploadUrl: string,
+  media: Media
+): Promise<UploadMediaResult> => {
+  if (Platform.OS === 'android') {
+    return await uploadMediaAndroid(client, uploadUrl, media)
+  } else if (Platform.OS === 'ios') {
+    return await uploadMediaIOS(client, uploadUrl, media)
+  }
```

Comment:
I don't know how much size it adds to install the 2 plugins and whether they are treeshaked or not, but I would have done it this way: 
- create an uploadMedia.ios.js file 
- creation of an uploadMedia.android.js file 

And then "import uploadMedias from ./uploadMedia" and let the bundler take care of itself.

Reply:
I just discovered platform specific extensions 😱 Very nice!